mpitests/test_adjoint_jacobian.py

```diff
@@ -0,0 +1,173 @@
+
+        grad_fn = qml.grad(qnode1)
+        grad_A = grad_fn(*args)
+
+        spy.assert_called()
+
+        qnode2 = QNode(circuit, dev_gpumpi, diff_method="parameter-shift")
+        grad_fn = qml.grad(qnode2)
+        grad_PS = grad_fn(*args)
+
+        assert np.allclose(grad_A, grad_PS, atol=tol, rtol=0)
+
+    def test_qnode_tensor_obs(self, mocker, tol):
+        """Test that specifying diff_method allows the adjoint method to be selected"""
+
+        num_wires = 3
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+        commSize = comm.Get_size()
+
+        args = np.array([0.54, 0.1, 0.5], requires_grad=True)
+
+        dev_gpumpi = qml.device("lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128)
+
+        def circuit(x, y, z):
+            qml.Hadamard(wires=0)
+            qml.RX(0.543, wires=0)
+            qml.CNOT(wires=[0, 1])
+
+            qml.Rot(x, y, z, wires=0)
+
+            qml.Rot(1.3, -2.3, 0.5, wires=[0])
+            qml.RZ(-0.5, wires=0)
+            qml.RY(0.5, wires=1)
+            qml.CNOT(wires=[0, 1])
+
+            return qml.expval(qml.PauliX(0) @ qml.PauliZ(1))
+
+        qnode1 = QNode(circuit, dev_gpumpi, diff_method="adjoint")
+        spy = mocker.spy(dev_gpumpi, "adjoint_jacobian")
+
+        grad_fn = qml.grad(qnode1)
+        grad_A = grad_fn(*args)
+
+        spy.assert_called()
+
+        qnode2 = QNode(circuit, dev_gpumpi, diff_method="parameter-shift")
+        grad_fn = qml.grad(qnode2)
+        grad_PS = grad_fn(*args)
+
+        assert np.allclose(grad_A, grad_PS, atol=tol, rtol=0)
+
+    def test_qnode_ham_obs(self, mocker, tol):
+        """Test that specifying diff_method allows the adjoint method to be selected"""
+
+        num_wires = 3
+        comm = MPI.COMM_WORLD
+        rank = comm.Get_rank()
+        commSize = comm.Get_size()
+
+        args = np.array([0.54, 0.1, 0.5], requires_grad=True)
+
+        dev_gpumpi = qml.device("lightning.gpu", wires=num_wires, mpi=True, c_dtype=np.complex128)
+
+        obs = qml.PauliX(1) @ qml.PauliY(2)
+        obs1 = qml.Identity(1)
+        H = qml.Hamiltonian([1.0, 1.0], [obs1, obs])
+
+        def circuit(x, y, z):
+            qml.Hadamard(wires=0)
+            qml.RX(0.543, wires=0)
+            qml.CNOT(wires=[0, 1])
+
+            qml.Rot(x, y, z, wires=0)
+
+            qml.Rot(1.3, -2.3, 0.5, wires=[0])
+            qml.RZ(-0.5, wires=0)
+            qml.RY(0.5, wires=1)
+            qml.CNOT(wires=[0, 1])
+
+            return qml.expval(H)
+
+        qnode1 = QNode(circuit, dev_gpumpi, diff_method="adjoint")
+        spy = mocker.spy(dev_gpumpi, "adjoint_jacobian")
+
+        grad_fn = qml.grad(qnode1)
+        grad_A = grad_fn(*args)
+
+        spy.assert_called()
+
+        qnode2 = QNode(circuit, dev_gpumpi, diff_method="parameter-shift")
+        grad_fn = qml.grad(qnode2)
+        grad_PS = grad_fn(*args)
+
+        assert np.allclose(grad_A, grad_PS, atol=tol, rtol=0)
```

Comment:
for any gates/observables that we know do not currently work we should also add tests that are expected failures too. This will make it easier to know when we changes things to support them later.